src/components/FilterSuggester/FilterSuggester.jsx

```diff
@@ -0,0 +1,306 @@
+      { name: 'comments', prefix: 'sort:' },
+      { name: 'reactions', prefix: 'sort:' },
+      { name: 'interactions', prefix: 'sort:' },
+      { name: 'reactions-+1-desc', prefix: 'sort:' },
+      { name: 'reactions--1-desc', prefix: 'sort:' },
+      { name: 'reactions-smile-desc', prefix: 'sort:' },
+      { name: 'reactions-thinking_face-desc', prefix: 'sort:' },
+      { name: 'reactions-heart-desc', prefix: 'sort:' },
+      { name: 'reactions-tada-desc', prefix: 'sort:' },
+      { name: 'reactions-+1', prefix: 'sort:' },
+      { name: 'reactions--1', prefix: 'sort:' },
+      { name: 'reactions-smile', prefix: 'sort:' },
+      { name: 'reactions-thinking_face', prefix: 'sort:' },
+      { name: 'reactions-heart', prefix: 'sort:' },
+      { name: 'reactions-tada', prefix: 'sort:' },
+    ],
+  },
+  { name: 'milestone:', tip: 'milestone name' },
+  { name: 'author:', tip: 'username' },
+  { name: 'assignee:', tip: 'username' },
+  { name: 'mentions:', tip: 'username' },
+  { name: 'commenter:', tip: 'username' },
+  { name: 'involves:', tip: 'username' },
+  { name: 'team:', tip: 'team name' },
+  { name: 'user:', tip: 'username' },
+  { name: 'org:', tip: 'organization name' },
+  { name: 'repo:', tip: 'user/repo or org/repo' },
+  { name: 'created:', tip: 'date' },
+  { name: 'updated:', tip: 'date' },
+  { name: 'closed:', tip: 'date' },
+  { name: 'merged:', tip: 'date' },
+  { name: 'head:', tip: 'branch name' },
+  { name: 'base:', tip: 'branch name' },
+  {
+    name: 'comments:',
+    tip: 'number',
+    suggestions: [
+      { name: '', prefix: 'comments:', tip: 'number' },
+      { name: '>', prefix: 'comments:', tip: 'number' },
+      { name: '<', prefix: 'comments:', tip: 'number' },
+      { name: '>=', prefix: 'comments:', tip: 'number' },
+      { name: '<=', prefix: 'comments:', tip: 'number' },
+    ],
+  },
+  {
+    name: 'reactions:',
+    tip: 'number',
+    suggestions: [
+      { name: '', prefix: 'reactions:', tip: 'number' },
+      { name: '>', prefix: 'reactions:', tip: 'number' },
+      { name: '<', prefix: 'reactions:', tip: 'number' },
+      { name: '>=', prefix: 'reactions:', tip: 'number' },
+      { name: '<=', prefix: 'reactions:', tip: 'number' },
+    ],
+  },
+  {
+    name: 'interactions:',
+    tip: 'number',
+    suggestions: [
+      { name: '', prefix: 'interactions:', tip: 'number' },
+      { name: '>', prefix: 'interactions:', tip: 'number' },
+      { name: '<', prefix: 'interactions:', tip: 'number' },
+      { name: '>=', prefix: 'interactions:', tip: 'number' },
+      { name: '<=', prefix: 'interactions:', tip: 'number' },
+    ],
+  },
+]
+
+class FilterSuggester extends React.Component {
+  constructor(props) {
+    super(props)
+    const value = props.value || ''
+    this.state = {
+      value,
+      previousValue: value,
+      suggestions: [],
+    }
+    this.onKeyPress = this.onKeyPress.bind(this)
+    this.onKeyUp = this.onKeyUp.bind(this)
+  }
```

Comment:
I'm using [this guy's technique](https://medium.com/@ericclemmons/react-event-preventdefault-78c28c950e46#.vtjap6b1c) of getting around the fact that events in your React handlers are not native events, thus calling `preventDefault` on them doesn't do anything. I needed to override pressing Enter and Esc so they wouldn't do the default action in the input field. Enter would normally submit the form, I want it to not submit the form if the user is just pressing Enter to apply a suggestion. Esc kept clearing out the field, and it still does but now I restore it immediately; I think the 'clear the field' logic is coming from the react-autocomplete package I'm using, but I couldn't figure out how to turn it off.

For using the native addEventListener and removeEventListener, the same function has to be passed to removeEventListener as was passed to addEventListener, otherwise, it doesn't get unbound. But we need the handlers to know what `this` is (namely the instance of FilterSuggester they're working on), so we have to do this `bind(this)` to tie them to that instance. That lets us just pass `this.onKeyPress` to addEventListener/removeEventListener, instead of creating an anonymous function like `e => this.onKeyPress(e)` as the parameter.